### PR TITLE
WINC-1005: Add optional Windows proxy jobs

### PR DESCRIPTION
**Pull request by saifshaikh48**
Adds a new optional job on master and release-4.14 branches
to test Windows on a vSphere cluster with OVN-hybrid networking
and a simple cluster-wide proxy enabled.

Ran `make update jobs`

### CODE REVIEW

**Comment by saifshaikh48** [comment]
/pj-rehearse pull-ci-openshift-windows-machine-config-operator-master-vsphere-proxy-e2e-operator pull-ci-openshift-windows-machine-config-operator-release-4.14-vsphere-proxy-e2e-operator

**Comment by saifshaikh48** [comment]
Cluster-wide Proxy resource exists within the test cluster. From must-gather cluster-scoped-resources/config.openshift.io:
```yaml
- apiVersion: config.openshift.io/v1
  kind: Proxy
  metadata:
    name: cluster
  spec:
    httpProxy: http://XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@XXXXXXXXXXXXX:3128
    httpsProxy: http://XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@XXXXXXXXXXXXX:3128
    noProxy: test.no-proxy.com
    trustedCA:
      name: ""
  status:
    httpProxy: http://XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@XXXXXXXXXXXXX:3128
    httpsProxy: http://XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@XXXXXXXXXXXXX:3128
    noProxy: .cluster.local,.svc,10.128.0.0/14,127.0.0.1,172.30.0.0/16,192.168.86.0/25,api-int.ci-op-zplfimdz-a3a04.vmc-ci.devcluster.openshift.com,localhost,test.no-proxy.com
```

In non-proxied clusters, the Proxy resource does not have these proxy vars/vals
```yaml
- apiVersion: config.openshift.io/v1
  kind: Proxy
  metadata:
    name: cluster
  spec:
    trustedCA:
      name: ""
  status: {}
```

File: ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml [hunk 1]
```diff
@@ -155,6 +155,22 @@ tests:
     test:
     - ref: windows-e2e-operator-test-byoh
     workflow: windows-e2e-operator-test-platform-none-vsphere
+- as: vsphere-proxy-e2e-operator
+  optional: true
```
Comment (sebsoto) [inline ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml:159]: `always_run` should be set to false as we expect these jobs to fail

Comment (saifshaikh48) [reply ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml:159]: They should actually pass. Windows nodes function as normal, just ignoring the proxy
[successful job example](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/37634/rehearse-37634-pull-ci-openshift-windows-machine-config-operator-master-vsphere-proxy-e2e-operator/1638979745913769984)

Comment (aravindhp) [reply ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml:159]: Why do we need to have this run on every PR until the feature is under development? We should be setting this `always_run` to `false`.

File: ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-release-4.14.yaml [hunk 2]
```diff
@@ -142,6 +142,22 @@ tests:
     test:
     - ref: windows-e2e-operator-test-byoh
     workflow: windows-e2e-operator-test-platform-none-vsphere
+- as: vsphere-proxy-e2e-operator
```
Comment (aravindhp) [inline ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-release-4.14.yaml:145]: I don't think you need to add this explicitly to `release-4.14`. The bot should automatically sync this.

**Review by aravindhp — CHANGES_REQUESTED** [review]

File: ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml (hunk 1 above)
Comment (saifshaikh48) [reply ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml:159]: Ah I misunderstood what `optional` did. I'll change this

**Comment by saifshaikh48** [comment]
/pj-rehearse pull-ci-openshift-windows-machine-config-operator-master-vsphere-proxy-e2e-operator

**Comment by saifshaikh48** [comment]
/pj-rehearse pull-ci-openshift-windows-machine-config-operator-master-vsphere-proxy-e2e-operator

**Review by aravindhp — APPROVED** [review]
Thanks for working on this, @saifshaikh48

**Comment by saifshaikh48** [comment]
/pj-rehearse ack

/hold
for one more lgtm

**Comment by alinaryan** [comment]
Thanks for working on this, 
/lgtm

Noticed the test is optional, so what types of situations should the test be triggered?

**Comment by saifshaikh48** [comment]
/unhold

Thanks for the lgtm @alinaryan 
> Noticed the test is optional, so what types of situations should the test be triggered?

We'd trigger this on PRs that deal with the proxy feature during development, and then after proxy is fully supported we can make this a mandatory test (or a periodic, but we can cross that bridge when we get there).

**Pull request merged**